### PR TITLE
feat: adds windows server 2022

### PR DESCRIPTION
**Pull request by nick5616**
**What this PR does / why we need it**:
Adds Windows Server 2022 to `Makefile`, `BASEIMAGE`, and `BASEIMAGE_CORE`
**Which issue(s) this PR fixes** *(optional, using `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when the PR gets merged)*:
Fixes #

**Special notes for your reviewer**:
_ltsc2022_ is the MCR tag for _Windows Server 2022_
https://hub.docker.com/_/microsoft-windows-nanoserver
There is a PR for the same thing here: kubernetes-sigs/azuredisk-csi-driver#1030

### CODE REVIEW

File: docker/BASEIMAGE [hunk 1]
```diff
@@ -4,3 +4,5 @@ windows/amd64/1809=mcr.microsoft.com/windows/nanoserver:1809
 windows/amd64/1903=mcr.microsoft.com/windows/nanoserver:1903
 windows/amd64/1909=mcr.microsoft.com/windows/nanoserver:1909
 windows/amd64/2004=mcr.microsoft.com/windows/nanoserver:2004
+windows/amd64/2004=mcr.microsoft.com/windows/nanoserver:ltsc2022
+
```
Comment (aramase) [inline docker/BASEIMAGE:8]: nit: remove extra new line

**Review by aramase — COMMENTED** [review]
Thank you for the PR. Just one small nit!

**Comment by nilekhc** [comment]
@aramase @tam7t We are not testing/validating these kind of changes anywhere. Should we add `make container-all` to our presubmit build job? Perhaps may be a separate presubmit job to validate docker build vs existing build job which validates code build. This way we can be sure of whatever arc/platform we are specifying or adding is always tested. WDYT?

**Comment by aramase** [comment]
> @aramase @tam7t We are not testing/validating these kind of changes anywhere. Should we add make container-all to our presubmit build job? Perhaps may be a separate presubmit job to validate docker build vs existing build job which validates code build. This way we can be sure of whatever arc/platform we are specifying or adding is always tested. WDYT?

@nilekhc `container-all` is used for windows test. We build the multi arch/os image and run tests against that for windows.

```bash
docker manifest inspect k8sprow.azurecr.io/driver:v1.0.0-e2e-d4245ff6
{
   "schemaVersion": 2,
   "mediaType": "application/vnd.docker.distribution.manifest.list.v2+json",
   "manifests": [
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 952,
         "digest": "sha256:0797874d8647fd1ac1f4d7039194cc877db5f37f27a4bdec9f204cadf220333f",
         "platform": {
            "architecture": "amd64",
            "os": "linux"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 952,
         "digest": "sha256:e89f2d904e3acdcb69e1f140f8a585cb6fd1fc5daa067e5814c154be43ba03ab",
         "platform": {
            "architecture": "arm64",
            "os": "linux"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 951,
         "digest": "sha256:51e95d8f7ec36e52b9d242baec871b605fef9c099f92433805d6b32ddf33f69b",
         "platform": {
            "architecture": "amd64",
            "os": "windows",
            "os.version": "10.0.17763.2183"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 951,
         "digest": "sha256:999f6dac5c4dd86b0423db3f9cc1e0017f357a8545b6a793d3699b1244457d5f",
         "platform": {
            "architecture": "amd64",
            "os": "windows",
            "os.version": "10.0.18362.1256"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 951,
         "digest": "sha256:20383a1cd645857c8a1f624ecfeb66dbc3adea71c0600019b197254f7e5f023b",
         "platform": {
            "architecture": "amd64",
            "os": "windows",
            "os.version": "10.0.18363.1556"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 951,
         "digest": "sha256:66fa6fd6dc996fe674d94f70e036ea021e7e4f1c63e080d8b0116892a346f8c9",
         "platform": {
            "architecture": "amd64",
            "os": "windows",
            "os.version": "10.0.19041.1237"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 951,
         "digest": "sha256:8940fd99597ba20e0c08aa97f24b5ecbe0b3fea23d455f383f5a6129d7dbf9a1",
         "platform": {
            "architecture": "amd64",
            "os": "windows",
            "os.version": "10.0.20348.230"
         }
      }
   ]
}
```

**Comment by nilekhc** [comment]
> > @aramase @tam7t We are not testing/validating these kind of changes anywhere. Should we add make container-all to our presubmit build job? Perhaps may be a separate presubmit job to validate docker build vs existing build job which validates code build. This way we can be sure of whatever arc/platform we are specifying or adding is always tested. WDYT?
> 
> @nilekhc `container-all` is used for windows test. We build the multi arch/os image and run tests against that for windows.
> 
> ```shell
> docker manifest inspect k8sprow.azurecr.io/driver:v1.0.0-e2e-d4245ff6
> {
>    "schemaVersion": 2,
>    "mediaType": "application/vnd.docker.distribution.manifest.list.v2+json",
>    "manifests": [
>       {
>          "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
>          "size": 952,
>          "digest": "sha256:0797874d8647fd1ac1f4d7039194cc877db5f37f27a4bdec9f204cadf220333f",
>          "platform": {
>             "architecture": "amd64",
>             "os": "linux"
>          }
>       },
>       {
>          "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
>          "size": 952,
>          "digest": "sha256:e89f2d904e3acdcb69e1f140f8a585cb6fd1fc5daa067e5814c154be43ba03ab",
>          "platform": {
>             "architecture": "arm64",
>             "os": "linux"
>          }
>       },
>       {
>          "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
>          "size": 951,
>          "digest": "sha256:51e95d8f7ec36e52b9d242baec871b605fef9c099f92433805d6b32ddf33f69b",
>          "platform": {
>             "architecture": "amd64",
>             "os": "windows",
>             "os.version": "10.0.17763.2183"
>          }
>       },
>       {
>          "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
>          "size": 951,
>          "digest": "sha256:999f6dac5c4dd86b0423db3f9cc1e0017f357a8545b6a793d3699b1244457d5f",
>          "platform": {
>             "architecture": "amd64",
>             "os": "windows",
>             "os.version": "10.0.18362.1256"
>          }
>       },
>       {
>          "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
>          "size": 951,
>          "digest": "sha256:20383a1cd645857c8a1f624ecfeb66dbc3adea71c0600019b197254f7e5f023b",
>          "platform": {
>             "architecture": "amd64",
>             "os": "windows",
>             "os.version": "10.0.18363.1556"
>          }
>       },
>       {
>          "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
>          "size": 951,
>          "digest": "sha256:66fa6fd6dc996fe674d94f70e036ea021e7e4f1c63e080d8b0116892a346f8c9",
>          "platform": {
>             "architecture": "amd64",
>             "os": "windows",
>             "os.version": "10.0.19041.1237"
>          }
>       },
>       {
>          "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
>          "size": 951,
>          "digest": "sha256:8940fd99597ba20e0c08aa97f24b5ecbe0b3fea23d455f383f5a6129d7dbf9a1",
>          "platform": {
>             "architecture": "amd64",
>             "os": "windows",
>             "os.version": "10.0.20348.230"
>          }
>       }
>    ]
> }
> ```

Got it. Tests are covered as part of `pull-secrets-store-csi-driver-e2e-windows` prow infra job. `kubetest` will build all the windows specific images and test them subsequently.

**Comment by nilekhc** [comment]
/lgtm

/hold
for other reviews..

File: docker/BASEIMAGE [hunk 2]
```diff
@@ -4,3 +4,4 @@ windows/amd64/1809=mcr.microsoft.com/windows/nanoserver:1809
 windows/amd64/1903=mcr.microsoft.com/windows/nanoserver:1903
 windows/amd64/1909=mcr.microsoft.com/windows/nanoserver:1909
 windows/amd64/2004=mcr.microsoft.com/windows/nanoserver:2004
+windows/amd64/2004=mcr.microsoft.com/windows/nanoserver:ltsc2022
```
Comment (nilekhc) [inline docker/BASEIMAGE:7]: nit: please add newline.

Earlier there were _two newlines_ and only one was expected to remove. EOF is expected to have a _newline_.

File: docker/BASEIMAGE_CORE [hunk 3]
```diff
@@ -2,3 +2,4 @@ windows/amd64/1809=gcr.io/k8s-staging-e2e-test-images/windows-servercore-cache:1
 windows/amd64/1903=gcr.io/k8s-staging-e2e-test-images/windows-servercore-cache:1.0-linux-amd64-1903
 windows/amd64/1909=gcr.io/k8s-staging-e2e-test-images/windows-servercore-cache:1.0-linux-amd64-1909
 windows/amd64/2004=gcr.io/k8s-staging-e2e-test-images/windows-servercore-cache:1.0-linux-amd64-2004
+windows/amd64/ltsc2022=gcr.io/k8s-staging-e2e-test-images/windows-servercore-cache:1.0-linux-amd64-ltsc2022
```
Comment (nilekhc) [inline docker/BASEIMAGE_CORE:5]: nit: please add newline.

**Comment by aramase** [comment]
/label tide/merge-method-squash

@nick5616 this is the label I was referring to for squash ^^

**Review by aramase — COMMENTED** [review]
/lgtm

**Comment by aramase** [comment]
I think we should also cherry pick this to `release-1.0` and include in the `v1.0.0` release. Thoughts?

**Comment by aramase** [comment]
@nick5616 Could you also cherry pick this PR to the `release-1.0` branch?

Run the following command and that'll create the PR automatically

```bash
export GITHUB_USER=<github user name>
hack/cherry_pick_pull.sh upstream/release-1.0 756
```

**Pull request merged**